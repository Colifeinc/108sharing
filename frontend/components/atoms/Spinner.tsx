/** @jsx jsx */
import * as React from 'react'
import { COLORS } from 'constants/colors'
import { css, jsx } from '@emotion/core'
import classNames from 'classnames'

interface IProps {
  className?: string
}

const spinnerStyle = css`
  .hollow-dots-spinner,
  .hollow-dots-spinner * {
    box-sizing: border-box;
  }

  .hollow-dots-spinner {
    height: 15px;
    width: calc(30px * 3);
  }

  .hollow-dots-spinner .dot {
    width: 15px;
    height: 15px;
    margin: 0 calc(15px / 2);
    background-color: ${COLORS.Primary};
    border-radius: 50%;
    float: left;
    transform: scale(0);
    animation: hollow-dots-spinner-animation 900ms ease infinite 0ms;
  }

  .hollow-dots-spinner .dot:nth-of-type(1) {
    animation-delay: calc(180ms * 1);
  }

  .hollow-dots-spinner .dot:nth-of-type(2) {
    animation-delay: calc(180ms * 2);
  }

  .hollow-dots-spinner .dot:nth-of-type(3) {
    animation-delay: calc(180ms * 3);
  }

  @keyframes hollow-dots-spinner-animation {
    50% {
      transform: scale(1);
      opacity: 1;
    }
    100% {
      opacity: 0;
    }
  }
`

// https://github.com/epicmaxco/epic-spinners/blob/master/src/components/lib/HollowDotsSpinner.vue
const Spinner: React.FC<IProps> = props => {
  return (
    <div
      className={classNames(['flex items-center justify-center', props.className])}
      css={spinnerStyle}
    >
      <div className="hollow-dots-spinner">
        <div className="dot" />
        <div className="dot" />
        <div className="dot" />
      </div>
    </div>
  )
}

export default Spinner
