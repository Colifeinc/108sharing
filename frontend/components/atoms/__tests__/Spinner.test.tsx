import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import * as React from 'react'
import Spinner from '../Spinner'

describe('atoms/Spinner.tsx', () => {
  it('should matches to the snapshot', () => {
    const wrapper = shallow(<Spinner />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
