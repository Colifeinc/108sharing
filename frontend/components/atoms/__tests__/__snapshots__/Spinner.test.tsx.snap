// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Spinner.tsx should matches to the snapshot 1`] = `
<EmotionCssPropInternal
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
  className="flex items-center justify-center"
  css={
    Object {
      "map": undefined,
      "name": "288pxk",
      "next": undefined,
      "styles": "
  .hollow-dots-spinner,
  .hollow-dots-spinner * {
    box-sizing: border-box;
  }

  .hollow-dots-spinner {
    height: 15px;
    width: calc(30px * 3);
  }

  .hollow-dots-spinner .dot {
    width: 15px;
    height: 15px;
    margin: 0 calc(15px / 2);
    background-color: #08C4B3;
    border-radius: 50%;
    float: left;
    transform: scale(0);
    animation: hollow-dots-spinner-animation 900ms ease infinite 0ms;
  }

  .hollow-dots-spinner .dot:nth-of-type(1) {
    animation-delay: calc(180ms * 1);
  }

  .hollow-dots-spinner .dot:nth-of-type(2) {
    animation-delay: calc(180ms * 2);
  }

  .hollow-dots-spinner .dot:nth-of-type(3) {
    animation-delay: calc(180ms * 3);
  }

  @keyframes hollow-dots-spinner-animation {
    50% {
      transform: scale(1);
      opacity: 1;
    }
    100% {
      opacity: 0;
    }
  }
",
      "toString": [Function],
    }
  }
>
  <div
    className="hollow-dots-spinner"
  >
    <div
      className="dot"
    />
    <div
      className="dot"
    />
    <div
      className="dot"
    />
  </div>
</EmotionCssPropInternal>
`;
